roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-synapse.conf.j2

```diff
@@ -87,6 +170,51 @@ server {
 			{% endfor %}
 		{% endif %}
 
+		{% if initial_sync_workers %}
+			{% for location in matrix_synapse_initial_sync_worker_locations %}
+			location ~ {{ location }} {
+				proxy_pass http://initial_sync_worker_upstream$request_uri;
+				proxy_set_header Host $host;
+			}
+			{% endfor %}
+		{% endif %}
+
+		{% if sync_workers %}
+			{% for location in matrix_synapse_sync_worker_locations %}
+			location ~ {{ location }} {
+				proxy_pass http://sync_worker_upstream$request_uri;
+				proxy_set_header Host $host;
+			}
+			{% endfor %}
+		{% endif %}
+
+		{% if presence_workers %}
+			{% for location in matrix_synapse_presence_worker_locations %}
+			location ~ {{ location }} {
+				proxy_pass http://presence_worker_upstream$request_uri;
+				proxy_set_header Host $host;
+			}
+			{% endfor %}
+		{% endif %}
+
+		{% if client_workers %}
+			{% for location in matrix_synapse_client_worker_locations %}
+			location ~ {{ location }} {
+				proxy_pass http://client_worker_upstream$request_uri;
+				proxy_set_header Host $host;
+			}
+			{% endfor %}
+		{% endif %}
+
+		{% if account_workers %}
+			{% for location in matrix_synapse_accounts_worker_locations %}
+			location ~ {{ location }} {
+				proxy_pass http://accounts_worker_upstream$request_uri;
```

Comment:
What's the meaning of this?

Reply:
Just added for flexibility if we want to try with a worker dedicated to account data endpoints.

Reply:
It is redirecting the request to the corresponding upstream (which are predefined above)